src/App.js

```diff
@@ -1,11 +1,52 @@
-import React from 'react';
+import React, {Component} from 'react';
+import {BrowserRouter as Router, Route, Switch, Redirect} from "react-router-dom";
+import './App.css';
+import {Login} from "../src/components/Login";
+import {Register} from "./components/Register";
+import {Welcome} from "./components/Welcome";
+import {Page404} from "./components/Page404";
+import {MainPage} from "./components/MainPage";
 
-function App() {
+const withAuth = (Component, auth) => {
+    return (props) => {
+        return <Component {...props} auth={auth}/>
+    };
+};
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const tokens = localStorage.getItem('AUTH');
+
+        this.state = {
+            tokens: tokens || null
+        };
+    }
+
+    auth = (tokens) => {
+        this.setState({tokens: tokens})
```

Comment:
You can move the logic of working with localStorage here. For example:
```
auth = (tokens) => {
        if (tokens) {
            localStorage.setItem(...)
        } else {
            localStorage.removeItem(...)
        }
        this.setState({tokens: tokens})
    }
```

In this case, other components will not need to know about localStorage and will simply use the `auth` method.